meson.build

```diff
@@ -80,12 +81,46 @@ configure_file(
     configuration: test_conf_data,
 )
 
-pytest = find_program('pytest', required: false)
-
-if pytest.found()
-    test('test-mctpd', pytest,
+tests = get_option('tests')
+if tests.allowed()
+    # The test suite contains integration-style tests that we wish to isolate.
+    # The tests drive mctpd's D-Bus interfaces and navigate relevant code paths
+    # by mocking kernel and MCTP device behaviour. The mock behaviours are
+    # implemented by encapsulating netlink and MCTP messages over Unix domain
+    # sockets bound to mock implementations in the test suite.
+    #
+    # Isolate the test suite under its own D-Bus session to prevent test
+    # behaviours impacting applications connected to the usual buses. To
+    # implement the isolation we need to work under some constraints:
+    #
+    # 1. mctpd's implementation connects to a bus by invoking `sd_bus_default()`,
+    #    which uses a combination of environment variables and session
+    #    properties to determine whether to connect to the system bus or the
+    #    session bus
+    #
+    # 2. dbus-run-session configures an isolated session bus and controls the
+    #    relevant environment variables for influencing the application to
+    #    connect to this isolated bus instance.
+    #
+    # `sd_bus_default()` selects the system bus if the application is
+    # not running in a systemd user slice. From experience, in a Github
+    # Action environment the system bus is selected by `sd_bus_default()`,
+    # implying that they aren't exploiting systemd slices. However,
+    # `sd_bus_default()` can be influenced to select the user session by setting
+    # `DBUS_STARTER_BUS_TYPE=user` in the environment. At this point with a
+    # naive approach we run up against 2, where `dbus-run-session` sanitises
+    # that configuration away
+    #
+    # Invoke pytest via a shell script under `dbus-run-session` so we can
+    # override the sanitation of `DBUS_STARTER_BUS_TYPE`, ensuring `test-mctpd`
+    # connects to the isolated session bus prepared by `dbus-run-session`.
+    pytest = find_program('pytest', required: tests)
+    script = 'export DBUS_STARTER_BUS_TYPE=user ; @0@ --tap'.format(pytest.full_path())
```

Comment:
Could we drop the extra `/bin/sh` step by setting the `env` of the `test()` call below?

Reply:
hm, apparently not:

dbus-run-session(1):

> The variables DBUS_SESSION_BUS_PID, DBUS_SESSION_BUS_WINDOWID, DBUS_STARTER_BUS_TYPE and DBUS_STARTER_ADDRESS are removed from the environment, if present.

Reply:
... might help if I read the comment above properly :D

Reply:
Can you unpack that thought a little more?

Does the suggestion account for the commentary between 105-116?

Reply:
haha, comment race!